File: Stack(LinkedList).cpp
#include <iostream>
using namespace std;
class Node{
    public:
    int data;
    Node* next;
    Node(){
        data=0;
        next=NULL;
    }
    Node(int data){
        this->data=data;
        this->next=NULL;
    }
};
class LinkedList{
    Node*top;
    public:
    LinkedList(){
        top=NULL;
    }
    void push(int data){
        Node *newNode = new Node(data);
        if(top== NULL){
            top=newNode;
            cout<<"Element Inserted"<<endl;
            return;
        }
        cout<<"Element Inserted"<<endl;
        newNode->next=this->top;
        this->top=newNode;
    }
    void display(){
        Node* temp = top;
        if(top==NULL){
            cout<<"Stack Underflow! ";
            return;
        }
        cout<<"Stack is: "<<endl;
        while(temp!=NULL){
            cout<<temp->data<<endl;
            temp=temp->next;
        }
    }
    void pop(){
        if(top==nullptr){
            cout<<"List is Empty";
        }
        Node*temp=top;
        top=top->next;
        delete temp;
        cout<<"Top Element removed";
    }
    void peek(){
        cout<<"The Top is: "<<top->data<<endl;
    }
};
int main(){
    int ch,data,n;
    LinkedList list;
    do{
        cout<<endl<<"Menu"<<endl;
        cout<<"1. Push "<<endl;
        cout<<"2. Pop "<<endl;
        cout<<"3. Display "<<endl;
        cout<<"4. Peek "<<endl;
        cout<<"5. Exit "<<endl;
        cout<<"Enter your choice: ";
        cin>>ch;
        switch(ch){
            case 1: cout<<"Enter the value to be inserted: ";cin>>data; list.push(data);break;
            case 2: list.pop();break;
            case 3: list.display();break;
            case 4: list.peek();break;
            case 5: cout<<"Menu exited!";break;
            default: cout<<"Invalid Case!"<<endl;
        }
    }
    while(ch!=5);
}

File: Stack(LinkedList).java
import java.util.Scanner;
public class Stack{
    class Node{
        int data;
        Node next;
        Node(){
            data=0;
            next=null;
        }
        Node(int data){
            this.data=data;
            next=null;
        }
    }
    Node top=null;
    public void Push(int data){
        Node newnode = new Node(data);
        if (top == null){
            top = newnode;
            System.out.println("Element Inserted");
        }
        else{
            newnode.next=this.top;
            this.top=newnode;
            System.out.println("Element Inserted");
        }
    }
    public void Pop(){
        if(top== null){
            System.out.println("Stack Underflow!");
        }
        Node temp =top;
        top=top.next;
        temp=null;
        System.out.println("Element Deleted!");
    }
    public void Display(){
        Node node = top;
        if(top==null){
            System.out.println("Stack Underflow!");
        }
        System.out.println("The Stack: ");
        while (node!= null){
            System.out.println(node.data);
            node = node.next;
        }
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Stack L = new Stack();
        int ch,data;
        do{
            System.out.println("Menu");
            System.out.println("1. Push ");
            System.out.println("2. Pop ");
            System.out.println("3. Display ");
            System.out.println("4. Exit ");
            System.out.println("Enter your choice: ");
            ch= sc.nextInt();
            switch (ch) {
                case 1:
                    System.out.println("Enter the Value to be Inserted: ");
                    data=sc.nextInt();
                    L.Push(data);
                    break;
                case 2:L.Pop();break;
                case 3: L.Display();break;
                case 4: System.out.println("Menu Exited");break;
                default: System.out.println("Invalid Choice ");
            }
        }
        while(ch!=4);
        sc.close();
    }   
}
